contracts/src/v0.8/vrf/dev/ArbitrumL1Fees.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {ArbGasInfo} from "../../vendor/@arbitrum/nitro-contracts/src/precompiles/ArbGasInfo.sol";
+
+/// @dev An abstract contract that provides Arbitrum specific L1 fee calculations.
+// solhint-disable-next-line contract-name-camelcase
+abstract contract ArbitrumL1Fees {
+  /// @dev ARBGAS_ADDR is the address of the ArbGasInfo precompile on Arbitrum.
+  /// @dev reference: https://github.com/OffchainLabs/nitro/blob/v2.0.14/contracts/src/precompiles/ArbGasInfo.sol#L10
+  address private constant ARBGAS_ADDR = address(0x000000000000000000000000000000000000006C);
+  ArbGasInfo private constant ARBGAS = ArbGasInfo(ARBGAS_ADDR);
+
+  /**
+   * @notice Returns the L1 fees that will be paid for the current transaction, given any calldata
+   * @notice for the current transaction. It uses ArbGas.getCurrentTxL1GasFees to get the fees.
+   * @notice On Arbitrum, the provided calldata is not used to calculate the fees.
```

Comment:
I'm pretty sure this is incorrect.

Reply:
While the calldata is not passed as an argument, I'm pretty sure it is used in the implementation under-the-hood.

L1 fees are proportional to calldata size, so I'm pretty sure it is impossible to get the L1 fee _without_ considering the calldata

Reply:
I copied it from ChainSpecificUtil to keep the changes at the minimum (that includes the comments as well).
This is what the official [docs](https://docs.arbitrum.io/build-decentralized-apps/precompiles/reference#arbgasinfo) say about this function: `GetCurrentTxL1GasFees gets the fee paid to the aggregator for posting this tx`
And this is where the calculation happens (if you dig in deeper you'll notice they're dissecting the whole tx): https://github.com/OffchainLabs/nitro/blob/master/arbos/tx_processor.go#L461

I guess it's probably for the best to remove this comment in every file that contains it? Stating that we use `ArbGas.getCurrentTxL1GasFees` to get the L1 fees should be more than enough.

Reply:
Fixed this code comment